src/Upgrader/Business/PackageManager/PackageManager.php

```diff
@@ -70,6 +81,32 @@ public function requireDev(PackageTransferCollection $packageCollection): Packag
         return $this->packageManagerClient->requireDev($packageCollection);
     }
 
+    /**
+     * @return \Upgrader\Business\PackageManager\Response\Collection\PackageManagerResponseCollection
+     */
+    public function update(): PackageManagerResponseCollection
+    {
+        $rawResponse = $this->packageManagerClient->update();
+
+        if ($rawResponse->isSuccess()) {
+            $updateResponseCollection = new PackageManagerResponseCollection();
+            $composerLockDiffResponseCollection = $this->composerLockDiffClient->getComposerLockDiff();
+
+            foreach ($composerLockDiffResponseCollection as $diffResponse) {
+                $updateResponse = new PackageManagerResponse(
+                    true,
+                    $rawResponse->getOutput(),
+                    $diffResponse->getPackageList(),
+                );
+                $updateResponseCollection->add($updateResponse);
+            }
+
+            return $updateResponseCollection;
+        }
+
+        return new PackageManagerResponseCollection([$rawResponse]);
```

Comment:
why we are creating twice new PackageManagerResponseCollection(),
we could move it out of the "if" statement

Reply:
Fixed